src/shadowbox/server/main.ts

```diff
@@ -65,34 +65,35 @@ async function main() {
 
   prometheus.collectDefaultMetrics({register: prometheus.register});
 
-  const proxyHostname = process.env.SB_PUBLIC_IP;
   // Default to production metrics, as some old Docker images may not have
   // SB_METRICS_URL properly set.
   const metricsCollectorUrl = process.env.SB_METRICS_URL || 'https://metrics-prod.uproxy.org';
   if (!process.env.SB_METRICS_URL) {
     logging.warn('process.env.SB_METRICS_URL not set, using default');
   }
 
-  if (!proxyHostname) {
-    logging.error('Need to specify SB_PUBLIC_IP for invite links');
-    process.exit(1);
-  }
-
-  logging.debug(`=== Config ===`);
-  logging.debug(`SB_PUBLIC_IP: ${proxyHostname}`);
-  logging.debug(`SB_METRICS_URL: ${metricsCollectorUrl}`);
-  logging.debug(`==============`);
+  const serverConfig =
+    server_config.readServerConfig(getPersistentFilename('shadowbox_server_config.json'));
 
   const DEFAULT_PORT = 8081;
-  const apiPortNumber = Number(process.env.SB_API_PORT || DEFAULT_PORT);
+  const apiPortNumber = Number(serverConfig.data().apiPort || process.env.SB_API_PORT || DEFAULT_PORT);
```

Comment:
Let's do this logic inside `readServerConfig()` https://github.com/Jigsaw-Code/outline-server/blob/72fb3e0ba52425d4126dd6264db54d18f1865ba5/src/shadowbox/server/server_config.ts#L49

This way we can migrate users to the config, away from the variable, on first run.

Reply:
Done